_episodes/14.md

```diff
@@ -0,0 +1,330 @@
+reorder and update a column without any expensive refactoring of tables or 
+moving data around and there will be no adverse effects on your data or
+metadata.
+
+Partition evolution and hidden partitions are particularly invaluable. Before
+in Hive, you had a very strict way you had to model consistent partitioning
+across an entire table. In Iceberg, the *partiion spec* is a description of how 
+to partition data in a table consisting of a list of source columns and
+transforms. Once the spec is created, it generates a partition tuple that is
+applied uniformly to the files created with that spec. Unlike Hive, that
+requires you to modify and send a special column that acts as the partiion
+value, Iceberg stores partition values unmodified. Here's an example partition
+spec generated in the Java API.
+
+```
+PartitionSpec spec = PartitionSpec.builderFor(schema)
+        .hour("event_time")
+        .identity("level")
+        .build();
+```
+
+In the example above, we create a separate hourly partition on the `event_time`
+field and use the `identity()` function on level to generate another level of
+partitioning on the `level` field. If at a later time, you decide you are
+getting too many small files because your partitions are too small, then you 
+can update the partition spec and Iceberg will start writing new files by the
+updated spec. Again, this is all without creating a new table and moving data
+around and all the queries return correctly. This kind of evolution has always 
+been a problem with Hive.
+
+<p align="center">
+  <img align="center" width="60%" height="100%" src="/assets/episode/14/partition-spec-evolution.png"/><br/>
+</p>
+
+If all that wasn't enough, you can also do time travel and version rollback with
+Iceberg. As we mentioned above, Iceberg keeps track of various snapshots of 
+your data in time through manifest files. As long as you keep those older 
+snapshots around, the files associated with those snapshots will stick around
+as well. This allows you to move around to previous views of the data which can
+come in so much handy for testing, recovery, and many other purposes. Just as
+you can time travel, you can make the time travel permanent by rolling back
+any unintended changes and deleting the undesired snapshot.
+
+Iceberg is also able to offer fast scan planning by filtering out the metadata
+files that are irrelevant to the scan, and using the partition spec to only find
+files containing responses to the data. Iceberg filters the metadata using
+partition value ranges and seeing if that is contained within the files of
+the metadata. Then while processing the list of manifest files, Iceberg will
+filter files by query predicates included in the partition, then apply column
+stats to help prune out files that don't match. Iceberg also uses multiple
+concurrent writers to speed things up as a final measure.
+
+Saving the best for last; Iceberg is a community standard and has 
```

Comment:
This repeats the spec content from earlier. Remove one of them or consolidate into one.